### DISCUSSION TITLE
Script to check for PRs that need rebasing

### DISCUSSION THREAD

**Discussion by tcharding** [rust-bitcoin/rust-bitcoin#1356] [Announcements]
I finally got around to writing a script to check all my open non-draft PRs that need rebasing, thought I'd share it.

https://github.com/tcharding/lang/blob/master/bash/rebase-pr.sh

PRs welcome :)

**Comment by tcharding** [comment]
Its still a bit noisy, I'd like to shut git up further when automerging but couldn't work that out.

**Comment by Kixunil** [comment]
This requires the GitHub CLI tool, right?

BTW found an interesting shell: [nushell](https://www.nushell.sh). Played with it a bit and it's really nice.

**Reply by tcharding** [reply]
Yes uses `gh` to interact with GitHub.

**Reply by tcharding** [reply]
`nushell` looks pretty cool, can you let me know if you get over the initial hurdle of differences and start using it as your daily shell please.

**Comment by apoelstra** [comment]
I have "setup bors" on my TODO list which would be able to do this, fwiw.

**Reply by tcharding** [reply]
I needed a solution today because I was wasting so much time going through open PRs to see which ones need rebasing (I routinely have > 20 PRs open). I'd love to see bors get set up though :)

**Comment by sanket1729** [comment]
Relatedly, I think we can use the [Darthbot](https://github.com/MarcoFalke/DrahtBot/blob/main/scripts/conflicts.py) conflicts script. This is the script that bitcoin core uses to add a bot comment to list PRs that would conflict if this PR is merged.

**Reply by tcharding** [reply]
Oh sweet, `DrahtBot` is just what I needed those months ago when I attempted to find a solution to conflicting open PRs.
